Sources/Shared/Models/SocialMediaPosts.swift

```diff
@@ -0,0 +1,175 @@
+
+    /// The identifier of the Facebook page being posted to.
+    public let destination: Int
+
+    /// The Facebook cateogry of the video.
+    public let categoryID: String
+
+    /// Whether or not this Facbeook post should be embeddable.
+    public let allowEmbedding: Bool
+
+    /// Whether or not this post should appear on the Facebook News Feed.
+    public let shouldAppearOnNewsFeed: Bool
+
+    /// Whether or not this video should be searchable and show up in the user's video library on Facebook.
+    public let isSecretVideo: Bool
+
+    /// Whether or not to allow social actions on the post on Facebook.
+    public let allowSocialActions: Bool
+
+    public init(
+        title: String,
+        description: String,
+        destination: Int,
+        categoryID: String,
+        allowEmbedding: Bool,
+        shouldAppearOnNewsFeed: Bool,
+        isSecretVideo: Bool,
+        allowSocialActions: Bool
+    ) {
+        self.title = title
+        self.description = description
+        self.destination = destination
+        self.categoryID = categoryID
+        self.allowEmbedding = allowEmbedding
+        self.shouldAppearOnNewsFeed = shouldAppearOnNewsFeed
+        self.isSecretVideo = isSecretVideo
+        self.allowSocialActions = allowSocialActions
+    }
+}
+
+/// A structure representing a single post to LinkedIn.
+public struct PublishToLinkedInPost {
+
+    /// The LinkedIn page identifier that the video will be posted to.
+    public let pageID: Int
+
+    /// The title of the post as it will appear on LinkedIn.
+    public let title: String
+
+    /// The description of the post as it will appear on LinkedIn.
+    public let description: String
+
+    public init(
+        pageID: Int,
+        title: String,
+        description: String
+    ) {
+        self.pageID = pageID
+        self.title = title
+        self.description = description
+    }
+}
+
+/// A structure representing a single tweet on Twitter.
+public struct PublishToTwitterPost {
+
+    /// The contents of the tweet as it will appear on Twitter.
+    public let tweet: String
+
+    public init(tweet: String) {
+        self.tweet = tweet
+    }
+}
+
+/// A structure representing a single post to YouTube.
+public struct PublishToYouTubePost {
+
+    /// The privacy options available when sharing a video to YouTube.
+    /// - public: The video will be publically available.
+    /// - private: The video will not be publically available, and only viewable to the owner.
+    public enum Privacy: String, CaseIterable {
+        case `public`
+        case `private`
+    }
+
+    /// The title of the video as it will appear on YouTube.
+    public let title: String
+
+    /// The description of the video as it will appear on YouTube.
+    public let description: String
+
+    /// A list of tags for the video on YouTube. May be nil.
+    public let tags: [String]?
+
+    /// The privacy option for this video on YouTube.
+    public let privacy: Privacy
```

Comment:
Unless we think there could be more privacy options down the line could maybe use `isPrivate` bool flag to simplify

Reply:
Good question. I checked with API and apparently there's a third option for `unlisted`, but it's not yet supported on the web front end. But it could at some point.

Reply:
Cool makes sense to keep as is